test/integration/test_httproute.go

```diff
@@ -190,3 +193,204 @@ func TestHTTPRouteV1Beta1(t *testing.T) {
+	deployment, err = GetEnv().Cluster().Client().AppsV1().Deployments(namespace.Name).Create(GetCtx(), deployment, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("exposing deployment %s via service", deployment.Name)
+	service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeClusterIP)
+	_, err = GetEnv().Cluster().Client().CoreV1().Services(namespace.Name).Create(GetCtx(), service, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	httpPort := gatewayv1.PortNumber(80)
+	pathMatchPrefix := gatewayv1.PathMatchPathPrefix
+	kindService := gatewayv1.Kind("Service")
+	pathPrefix := "/prefix-test-http-route"
+	httpRoute := &gatewayv1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace.Name,
+			Name:      uuid.NewString(),
+			Annotations: map[string]string{
+				"konghq.com/strip-path": "true",
+			},
+		},
+		Spec: gatewayv1.HTTPRouteSpec{
+			CommonRouteSpec: gatewayv1.CommonRouteSpec{
+				ParentRefs: []gatewayv1.ParentReference{{
+					Name: gatewayv1.ObjectName(gateway.Name),
+				}},
+			},
+			Rules: []gatewayv1.HTTPRouteRule{
+				{
+					Matches: []gatewayv1.HTTPRouteMatch{
+						{
+							Path: &gatewayv1.HTTPPathMatch{
+								Type:  &pathMatchPrefix,
+								Value: &pathPrefix,
+							},
+						},
+					},
+					BackendRefs: []gatewayv1.HTTPBackendRef{
+						{
+							BackendRef: gatewayv1.BackendRef{
+								BackendObjectReference: gatewayv1.BackendObjectReference{
+									Name: gatewayv1.ObjectName(service.Name),
+									Port: &httpPort,
+									Kind: &kindService,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	t.Logf("creating httproute %s/%s to access deployment %s via kong", httpRoute.Namespace, httpRoute.Name, deployment.Name)
+	require.EventuallyWithT(t,
+		func(c *assert.CollectT) {
+			result, err := GetClients().GatewayClient.GatewayV1().HTTPRoutes(namespace.Name).Create(GetCtx(), httpRoute, metav1.CreateOptions{})
+			if err != nil {
+				t.Logf("failed to deploy httproute: %v", err)
+				c.Errorf("failed to deploy httproute: %v", err)
+				return
+			}
+			cleaner.Add(result)
+		},
+		testutils.DefaultIngressWait, testutils.WaitIngressTick,
+	)
+
+	t.Log("verifying connectivity to the HTTPRoute")
+	const (
+		httpRouteAccessTimeout = 3 * time.Minute
+		waitTick               = time.Second
+	)
+
+	require.Eventually(
+		t, testutils.GetResponseBodyContains(
+			t, GetCtx(), clients, httpsClient, "https://"+gatewayIPAddress+"/prefix-test-http-route", http.MethodGet, "<title>httpbin.org</title>",
+		),
+		httpRouteAccessTimeout, time.Second,
+	)
+	// will route to path /1234 of service httpbin, but httpbin will return a 404 page on this path.
+	require.Eventually(
+		t, testutils.GetResponseBodyContains(
+			t, GetCtx(), clients, httpsClient, "https://"+gatewayIPAddress+"/prefix-test-http-route/1234", http.MethodGet, "<h1>Not Found</h1>",
+		),
+		httpRouteAccessTimeout, time.Second,
+	)
```

Comment:
Ditto for this part - `TestHTTPRoute` could share common helpers with `TestHTTPRouteWithTLS`.